src/main/java/com/cleanroommc/groovyscript/compat/mods/aetherlegacy/Accessory.java

```diff
@@ -0,0 +1,88 @@
+package com.cleanroommc.groovyscript.compat.mods.aetherlegacy;
+
+import com.cleanroommc.groovyscript.api.GroovyLog;
+import com.cleanroommc.groovyscript.api.documentation.annotations.*;
+import com.cleanroommc.groovyscript.helper.Alias;
+import com.cleanroommc.groovyscript.helper.recipe.AbstractRecipeBuilder;
+import com.cleanroommc.groovyscript.registry.ForgeRegistryWrapper;
+import com.cleanroommc.groovyscript.registry.ReloadableRegistryManager;
+import com.gildedgames.the_aether.api.accessories.AccessoryType;
+import com.gildedgames.the_aether.api.accessories.AetherAccessory;
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.fml.common.registry.GameRegistry;
+import org.jetbrains.annotations.Nullable;
+
+@RegistryDescription
+public class Accessory extends ForgeRegistryWrapper<AetherAccessory> {
+
+    private static final AccessoryType[] VALUES = AccessoryType.values();
+
+    public Accessory() {
+        super(GameRegistry.findRegistry(AetherAccessory.class), Alias.generateOf("Accessory"));
+    }
+
+    @MethodDescription(type = MethodDescription.Type.ADDITION)
+    public void add(ItemStack item, String type) {
+        AccessoryType accessoryType = AccessoryType.MISC;
+        for (AccessoryType value : VALUES) {
+            if (value.getDisplayName().equalsIgnoreCase(type)) {
+                accessoryType = value;
+            }
+        }
+        AetherAccessory accessory = new AetherAccessory(item, accessoryType);
+        add(accessory);
+    }
+
+    @MethodDescription(description = "groovyscript.wiki.removeByInput", example = @Example("item('aether_legacy:iron_pendant')"))
+    public void removeByInput(ItemStack input) {
+        this.getRegistry().getValuesCollection().forEach(accessory -> {
+            if (accessory.getAccessoryStack().isItemEqual(input)) {
+                ReloadableRegistryManager.removeRegistryEntry(this.getRegistry(), accessory.getRegistryName());
+            }
```

Comment:
Rather use `input.test(accessory.getAccessoryStack())` in the if. With that you can replace the `ItemStack` with `IIngredient`. And to remove the recipe you can just call `remove(accessory).` And maybe even log an error if no recipes was found or the input is empty (check with `IngredientHelper.isEmpty(input)`).
```suggestion
public void removeByInput(IIngredient input) {
        this.getRegistry().getValuesCollection().forEach(accessory -> {
            if (input.test(accessory.getAccessoryStack())) {
                remove(accessory);
            }
        });;
```